tests/test_cli_options.py

```diff
@@ -0,0 +1,507 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from distutils.cmd import Command
+from email.policy import default
+import os
+import copy
+
+from model_analyzer.config.input.config_defaults import DEFAULT_TRITON_DOCKER_IMAGE
+
+from .common import test_result_collector as trc
+
+from model_analyzer.cli.cli import CLI
+from model_analyzer.config.input.config_command_profile import ConfigCommandProfile
+from model_analyzer.config.input.config_command_analyze import ConfigCommandAnalyze
+from model_analyzer.config.input.config_command_report import ConfigCommandReport
+from model_analyzer.config.input.config_status import ConfigStatus
+from model_analyzer.constants import CONFIG_PARSER_SUCCESS
+
+import psutil
+
+from unittest.mock import patch
+
+
+class CLISubclass(CLI):
+    """
+    Subclass of CLI to overwrite the parse method.
+    Parse takes a list of arguments instead of getting the args
+    from sys.argv
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def parse(self, parsed_commands=None):
+        args = self._parser.parse_args(parsed_commands[1:])
+        if args.subcommand is None:
+            self._parser.print_help()
+            self._parser.exit()
+        config = self._subcommand_configs[args.subcommand]
+        config.set_config_values(args)
+        return args, config
+
+
+class CLIConfigProfileStruct():
+    """
+    Struct class to hold the common variables shared between profile tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'profile',
+            '--model-repository',
+            'foo',
+            '--profile-models',
+            'bar'
+        ]
+        #yapf: enable
+        config_profile = ConfigCommandProfile()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='profile', help='', config=config_profile)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class CLIConfigAnalyzeStruct():
+    """
+    Struct class to hold the common variables shared between analyze tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'analyze',
+            '--analysis-models',
+            'a,b,c'
+        ]
+        #yapf: enable
+        config_analyze = ConfigCommandAnalyze()
+        self.cli = CLISubclass()
+        self.cli.add_subcommand(cmd='analyze', help='', config=config_analyze)
+
+    def parse(self):
+        return self.cli.parse(self.args)
+
+
+class CLIConfigReportStruct():
+    """
+    Struct class to hold the common variables shared between analyze tests
+    """
+
+    def __init__(self):
+        #yapf: disable
+        self.args = [
+            '/usr/local/bin/model-analyzer',
+            'report',
+            '--report-model-configs',
+            'a, b, c'
```

Comment:
Is there any reason this has spaces after the commas and analyze doesn't?

Reply:
Nope. They should both work